stackit/internal/services/iaas/network/resource.go

```diff
@@ -450,59 +585,113 @@ func toCreatePayload(ctx context.Context, model *Model) (*iaas.CreateNetworkPayl
 	if model == nil {
 		return nil, fmt.Errorf("nil model")
 	}
+	addressFamily := &iaas.CreateNetworkAddressFamily{}
+
+	modelIPv6Nameservers := []string{}
+	for _, ipv6ns := range model.IPv6Nameservers.Elements() {
+		ipv6NameserverString, ok := ipv6ns.(types.String)
+		if !ok {
+			return nil, fmt.Errorf("type assertion failed")
+		}
+		modelIPv6Nameservers = append(modelIPv6Nameservers, ipv6NameserverString.ValueString())
+	}
+
+	if !(model.IPv6Prefix.IsNull() || model.IPv6PrefixLength.IsNull() || model.IPv6Nameservers.IsNull()) {
+		addressFamily.Ipv6 = &iaas.CreateNetworkIPv6Body{
+			Nameservers:  &modelIPv6Nameservers,
+			Gateway:      iaas.NewNullableString(conversion.StringValueToPointer(model.IPv6Gateway)),
+			Prefix:       conversion.StringValueToPointer(model.IPv6Prefix),
+			PrefixLength: conversion.Int64ValueToPointer(model.IPv6PrefixLength),
+		}
+	}
 
-	modelNameservers := []string{}
-	for _, ns := range model.Nameservers.Elements() {
-		nameserverString, ok := ns.(types.String)
+	modelIPv4Nameservers := []string{}
+	for _, ipv4ns := range model.IPv4Nameservers.Elements() {
+		ipv4NameserverString, ok := ipv4ns.(types.String)
 		if !ok {
 			return nil, fmt.Errorf("type assertion failed")
 		}
-		modelNameservers = append(modelNameservers, nameserverString.ValueString())
+		modelIPv4Nameservers = append(modelIPv4Nameservers, ipv4NameserverString.ValueString())
+	}
+
+	if !(model.IPv4Prefix.IsNull() || model.IPv4PrefixLength.IsNull() || model.IPv4Nameservers.IsNull()) {
+		addressFamily.Ipv4 = &iaas.CreateNetworkIPv4Body{
+			Nameservers:  &modelIPv4Nameservers,
+			Gateway:      iaas.NewNullableString(conversion.StringValueToPointer(model.IPv4Gateway)),
+			Prefix:       conversion.StringValueToPointer(model.IPv4Prefix),
+			PrefixLength: conversion.Int64ValueToPointer(model.IPv4PrefixLength),
+		}
 	}
```

Comment:
We should continue supporting the `nameservers` field while it's deprecated. Currently this will be a breaking change for users using the `nameservers`

Reply:
So we should use the `IPv4Nameservers` if provided, otherwise fallback to `nameservers`

Reply:
And the same for the update

Reply:
We can also a test case for this in the unit tests